cmd/crc/cmd/daemon.go

```diff
@@ -142,7 +143,9 @@ func run(configuration *types.Configuration) error {
 		}
 		mux := http.NewServeMux()
 		mux.Handle("/network/", http.StripPrefix("/network", vn.Mux()))
-		mux.Handle("/api/", http.StripPrefix("/api", api.NewMux(config, newMachine(), logging.Memory, segmentClient)))
+		machineClient := newMachine()
+		mux.Handle("/api/", http.StripPrefix("/api", api.NewMux(config, machineClient, logging.Memory, segmentClient)))
+		mux.Handle("/socket/", http.StripPrefix("/socket", websocket.NewWebsocketServer(machineClient)))
```

Comment:
wondering if this could've been part of the same `/api` path, something like `/api/statuswatch` ?

Reply:
Hm, It may, let me try.

I prefer to separate HTTP API and WebSocket API, in URL. We may have only one websocket endpoint, and add some soer of PRC API or PubSub  API, and handle all events/query’s in single websocket connection. But this approach requires more complex client implementation.

Reply:
okay, don't change things yet if it complicates the client implementations! i was just asking to know if it was possible or there was some limitations to using the same HTTP server for websocket as well as normal HTTP